wix-embedded-mysql/src/main/java/com/wix/mysql/distribution/setup/NixBefore8Initializer.java

```diff
@@ -0,0 +1,39 @@
+package com.wix.mysql.distribution.setup;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.commons.io.filefilter.RegexFileFilter;
+
+import com.wix.mysql.config.MysqldConfig;
+import com.wix.mysql.distribution.Version;
+
+import de.flapdoodle.embed.process.config.IRuntimeConfig;
+import de.flapdoodle.embed.process.distribution.Platform;
+import de.flapdoodle.embed.process.extract.IExtractedFileSet;
+
+public class NixBefore8Initializer implements Initializer {
+
+        private static final String SEP = File.separator;
+    
+    @Override
+    public boolean matches(Version version) {
+        return Platform.detect().isUnixLike() &&
+                version.getMajorVersion().equals("8.0");
+    }
+
+    @Override
+    public void apply(IExtractedFileSet files, IRuntimeConfig runtimeConfig, MysqldConfig config) throws IOException {
+        File baseDir = files.baseDir();
+        File libDir = new File(baseDir + SEP + "lib");
+        FileFilter filter = new RegexFileFilter("^[a-z]+\\.so(\\.[0-9]+)+");
+        File[] soFiles = libDir.listFiles(filter);
+        for (File file : soFiles) {
+            Files.createSymbolicLink(Paths.get(baseDir + SEP + "bin" + SEP + file.getName()), Paths.get(file.getPath()));
```

Comment:
I definitely like this much better than the original implementation. ^_^

Can you also enhance this part to support 8.0.19 and later's style?
```
$ pwd; ls -lR lib/*/*.so.*
mysql-8.0.19-linux-glibc2.12-x86_64

-rwxr-xr-x 1 esun eng  3056988 Dec  9 14:43 lib/private/libcrypto.so.1.1
-rwxr-xr-x 1 esun eng  5408408 Dec  9 14:43 lib/private/libprotobuf-lite.so.3.6.1
-rwxr-xr-x 1 esun eng 45294256 Dec  9 14:45 lib/private/libprotobuf.so.3.6.1
-rwxr-xr-x 1 esun eng   657635 Dec  9 14:43 lib/private/libssl.so.1.1
```
The 4 so files are all needed, but they are one level deeper inside `lib`

Reply:
The master readme doc specifically states the req for those OS level dependencies to be installed. This will work instead of those deps? If so, I  think that should be a different patch to remove the external dependencies.

Reply:
The master readme approach will be pretty annoying for CI/INT environment, because we don't have permission to go there and manually install any 3rd party files. For Linux and Mac, if we can make use of the bundled *.so files, why do we have to sudo install anything?

As for Windows, the DDL files are all inside the **bin** already:
```
$ ls -l mysql-8.0.20-winx64/bin/*.dll
-rw-rw-rw- 1 esun eng 3384320 Mar  6 04:21 mysql-8.0.20-winx64/bin/libcrypto-1_1-x64.dll
-rw-rw-rw- 1 esun eng 1839616 Feb 27 04:46 mysql-8.0.20-winx64/bin/libmecab.dll
-rw-rw-rw- 1 esun eng 6380032 Mar 31 05:04 mysql-8.0.20-winx64/bin/libprotobuf-debug.dll
-rw-rw-rw- 1 esun eng 2573312 Mar 31 05:20 mysql-8.0.20-winx64/bin/libprotobuf.dll
-rw-rw-rw- 1 esun eng 1039872 Mar 31 05:03 mysql-8.0.20-winx64/bin/libprotobuf-lite-debug.dll
-rw-rw-rw- 1 esun eng  404992 Mar 31 05:18 mysql-8.0.20-winx64/bin/libprotobuf-lite.dll
-rw-rw-rw- 1 esun eng  679424 Mar  6 04:21 mysql-8.0.20-winx64/bin/libssl-1_1-x64.dll
```